# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: 0132624b39cf
# Full Hash: 0132624b39cf64e59e43b737b2126cc451e39ca6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-12-19 21:52:02
# Description:
#   Bug 1603767 - Ensure that NoteActorDestroyed is not called multiple times. r=dom-workers-and-storage-reviewers,janv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D57301
# ==============================================================================

diff -r fa9637bf7ba6 -r 0132624b39cf dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Thu Dec 19 07:58:57 2019 +0000
+++ b/dom/indexedDB/ActorsParent.cpp	Thu Dec 19 09:55:05 2019 +0000
@@ -21683,7 +21683,14 @@
   // A bit hacky but the VersionChangeOp is not generated in response to a
   // child request like most other database operations. Do this to make our
   // assertions happy.
-  NoteActorDestroyed();
+  //
+  // XXX: Depending on timing, in most cases, NoteActorDestroyed will not have
+  // been destroyed before, but in some cases it has. This should be reworked in
+  // a way this hack is not necessary. There are also several similar cases in
+  // other *Op classes.
+  if (!IsActorDestroyed()) {
+    NoteActorDestroyed();
+  }
 #endif
 
   TransactionDatabaseOperationBase::Cleanup();
